store/gaskv/store.go

```diff
@@ -177,8 +172,11 @@ func (gi *gasIterator) Error() error {
 // consumeSeekGas consumes on each iteration step a flat gas cost and a variable gas cost
 // based on the current value's length.
 func (gi *gasIterator) consumeSeekGas() {
-	value := gi.Value()
+	if gi.Valid() {
+		value := gi.Value()
 
-	gi.gasMeter.ConsumeGas(gi.gasConfig.ReadCostPerByte*types.Gas(len(value)), types.GasValuePerByteDesc)
-	gi.gasMeter.ConsumeGas(gi.gasConfig.IterNextCostFlat, types.GasIterNextCostFlatDesc)
+		gi.gasMeter.ConsumeGas(gi.gasConfig.ReadCostPerByte*types.Gas(len(value)), types.GasValuePerByteDesc)
+		gi.gasMeter.ConsumeGas(gi.gasConfig.IterNextCostFlat, types.GasIterNextCostFlatDesc)
```

Comment:
The issue's proposal is actually to always charge IterNextCostFlat. Then, if the value is valid, we additionally charge ReadCostPerByte.

So L179 should go outside of the if

Reply:
oh, thanks amaury